ebos/eclwriter.hh

```diff
@@ -190,15 +189,34 @@ public:
 
         this->captureLocalFluxData();
 
-        if (this->collectToIORank_.isParallel())
+        if (this->collectToIORank_.isParallel()) {
+            OPM_BEGIN_PARALLEL_TRY_CATCH()
+
             this->collectToIORank_.collect({},
                                            eclOutputModule_->getBlockData(),
                                            eclOutputModule_->getWBPData(),
                                            localWellData,
                                            localGroupAndNetworkData,
                                            localAquiferData,
-                                           localWellTestState);
+                                           localWellTestState,
+                                           this->eclOutputModule_->getInterRegFlows());
+
+            if (this->collectToIORank_.isIORank()) {
+                auto& iregFlows = this->collectToIORank_.globalInterRegFlows();
+
+                if (! iregFlows.readIsConsistent()) {
+                    throw std::runtime_error {
+                        "Failed to correctly collect all "
+                        "inter-region flows on I/O rank"
```

Comment:
Why not  tell what the underlying problem is? "Experienced inconsistent inter-region flow names in parallel". Maybe `logic_error` is more suitable?

Reply:
> Why not tell what the underlying problem is?

Good idea.  I'll update.

> Maybe `logic_error` is more suitable?

Maybe.  I think it's a tossup in this particular case since `runtime_error` are

> errors that are due to events beyond the scope of the program and can not be easily predicted.

([cppreference.com](https://en.cppreference.com/w/cpp/error/runtime_error)).  Especially the notion of "not easily predicted" is applicable here because if we ever actually fail in this way then there are events beyond our control that cannot be predicted.

Reply:
Well, when stubborn smart-asses bounce their heads together... :wink:  I guess we just can't help it.

I think this is a misinterpretation. They are talking about the events that cause the exception (which is clearly faulty program logic here). Take a look at the derived classes. You are saying that the exception choice should be governed by events that could occur if an exception is not thrown.

Let's not waste time, I will merge tonight either way.